packages/toolkit/src/query/core/buildMiddleware/invalidationByTags.ts

```diff
@@ -55,12 +71,27 @@ export const buildInvalidationByTagsHandler: InternalHandlerBuilder = ({
   }
 
   function invalidateTags(
-    tags: readonly FullTagDescription<string>[],
+    newTags: readonly FullTagDescription<string>[],
     mwApi: SubMiddlewareApi
   ) {
     const rootState = mwApi.getState()
     const state = rootState[reducerPath]
 
+    pendingTagInvalidations.push(...newTags)
+
+    if (state.config.invalidationBehavior === 'delayed') {
+      const hasPendingRequests = [
```

Comment:
💬 I have a partial concern here that calling `Object.values()` isn't entirely efficient. Might be an over-micro-optimization concern, but I could foresee that with a lot of queries this could get vaguely expensive.  (Only immediate alternative I have is some `for...in` loops, though.)

Reply:
Will fix